### PR TITLE
CLIENTS-1659: Remove V1 API Kafka core dependency and associated Zookeeper Client configs

### PR DESCRIPTION
**Pull request by ijuma**
We should only depend on the kafka-clients jar.

### CODE REVIEW

**Comment by rnpridgeon** [comment]
[INFO] --- maven-dependency-plugin:3.1.2:tree (default-cli) @ kafka-rest ---
[INFO] io.confluent:kafka-rest:jar:6.0.0-SNAPSHOT
[INFO] +- com.github.spotbugs:spotbugs-annotations:jar:4.0.2:compile
[INFO] |  \- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] +- com.google.auto.value:auto-value-annotations:jar:1.7.2:compile
[INFO] +- io.confluent:rest-utils:jar:6.0.0-SNAPSHOT:compile
[INFO] |  +- org.eclipse.jetty.websocket:javax-websocket-server-impl:jar:9.4.24.v20191120:compile
[INFO] |  |  +- org.eclipse.jetty:jetty-annotations:jar:9.4.24.v20191120:compile
[INFO] |  |  |  +- org.eclipse.jetty:jetty-plus:jar:9.4.24.v20191120:compile
[INFO] |  |  |  |  \- org.eclipse.jetty:jetty-jndi:jar:9.4.24.v20191120:compile
[INFO] |  |  |  +- org.eclipse.jetty:jetty-webapp:jar:9.4.24.v20191120:compile
[INFO] |  |  |  |  \- org.eclipse.jetty:jetty-xml:jar:9.4.24.v20191120:compile
[INFO] |  |  |  +- javax.annotation:javax.annotation-api:jar:1.3:compile
[INFO] |  |  |  +- org.ow2.asm:asm:jar:7.2:compile
[INFO] |  |  |  \- org.ow2.asm:asm-commons:jar:7.2:compile
[INFO] |  |  |     +- org.ow2.asm:asm-tree:jar:7.2:compile
[INFO] |  |  |     \- org.ow2.asm:asm-analysis:jar:7.2:compile
[INFO] |  |  +- org.eclipse.jetty.websocket:javax-websocket-client-impl:jar:9.4.24.v20191120:compile
[INFO] |  |  |  +- org.eclipse.jetty.websocket:websocket-client:jar:9.4.24.v20191120:compile
[INFO] |  |  |  |  \- org.eclipse.jetty:jetty-client:jar:9.4.24.v20191120:compile
[INFO] |  |  |  \- javax.websocket:javax.websocket-client-api:jar:1.0:compile
[INFO] |  |  +- org.eclipse.jetty.websocket:websocket-server:jar:9.4.24.v20191120:compile
[INFO] |  |  |  +- org.eclipse.jetty.websocket:websocket-common:jar:9.4.24.v20191120:compile
[INFO] |  |  |  |  \- org.eclipse.jetty.websocket:websocket-api:jar:9.4.24.v20191120:compile
[INFO] |  |  |  \- org.eclipse.jetty.websocket:websocket-servlet:jar:9.4.24.v20191120:compile
[INFO] |  |  \- javax.websocket:javax.websocket-api:jar:1.0:compile
[INFO] |  +- org.glassfish.jersey.containers:jersey-container-servlet:jar:2.30:compile
[INFO] |  |  +- org.glassfish.jersey.containers:jersey-container-servlet-core:jar:2.30:compile
[INFO] |  |  +- org.glassfish.jersey.core:jersey-common:jar:2.30:compile
[INFO] |  |  |  +- jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile
[INFO] |  |  |  \- org.glassfish.hk2:osgi-resource-locator:jar:1.0.3:compile
[INFO] |  |  +- org.glassfish.jersey.core:jersey-server:jar:2.30:compile
[INFO] |  |  |  +- org.glassfish.jersey.core:jersey-client:jar:2.30:compile
[INFO] |  |  |  \- org.glassfish.jersey.media:jersey-media-jaxb:jar:2.30:compile
[INFO] |  |  \- jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile
[INFO] |  +- org.glassfish.jersey.inject:jersey-hk2:jar:2.30:compile
[INFO] |  |  +- org.glassfish.hk2:hk2-locator:jar:2.6.1:compile
[INFO] |  |  |  +- org.glassfish.hk2.external:aopalliance-repackaged:jar:2.6.1:compile
[INFO] |  |  |  +- org.glassfish.hk2:hk2-api:jar:2.6.1:compile
[INFO] |  |  |  \- org.glassfish.hk2:hk2-utils:jar:2.6.1:compile
[INFO] |  |  \- org.javassist:javassist:jar:3.25.0-GA:compile
[INFO] |  +- javax.xml.bind:jaxb-api:jar:2.3.0:compile
[INFO] |  +- javax.activation:activation:jar:1.1.1:compile
[INFO] |  +- org.glassfish.jersey.ext:jersey-bean-validation:jar:2.30:compile
[INFO] |  |  +- org.glassfish.hk2.external:jakarta.inject:jar:2.6.1:compile
[INFO] |  |  +- jakarta.validation:jakarta.validation-api:jar:2.0.2:compile
[INFO] |  |  +- org.hibernate.validator:hibernate-validator:jar:6.0.17.Final:compile
[INFO] |  |  |  +- org.jboss.logging:jboss-logging:jar:3.3.2.Final:compile
[INFO] |  |  |  \- com.fasterxml:classmate:jar:1.3.4:compile
[INFO] |  |  +- jakarta.el:jakarta.el-api:jar:3.0.3:compile
[INFO] |  |  \- org.glassfish:jakarta.el:jar:3.0.2:compile
[INFO] |  +- org.eclipse.jetty:jetty-jmx:jar:9.4.24.v20191120:compile
[INFO] |  |  \- org.eclipse.jetty:jetty-util:jar:9.4.24.v20191120:compile
[INFO] |  +- org.eclipse.jetty:jetty-server:jar:9.4.24.v20191120:compile
[INFO] |  |  +- javax.servlet:javax.servlet-api:jar:3.1.0:compile
[INFO] |  |  +- org.eclipse.jetty:jetty-http:jar:9.4.24.v20191120:compile
[INFO] |  |  \- org.eclipse.jetty:jetty-io:jar:9.4.24.v20191120:compile
[INFO] |  +- org.eclipse.jetty:jetty-servlet:jar:9.4.24.v20191120:compile
[INFO] |  |  \- org.eclipse.jetty:jetty-security:jar:9.4.24.v20191120:compile
[INFO] |  +- org.eclipse.jetty:jetty-servlets:jar:9.4.24.v20191120:compile
[INFO] |  |  \- org.eclipse.jetty:jetty-continuation:jar:9.4.24.v20191120:compile
[INFO] |  +- org.eclipse.jetty:jetty-jaas:jar:9.4.24.v20191120:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.10.5:compile
[INFO] |  +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.10.5:compile
[INFO] |  |  \- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.10.5:compile
[INFO] |  |     +- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile
[INFO] |  |     \- jakarta.activation:jakarta.activation-api:jar:1.2.1:compile
[INFO] |  +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.10.5:compile
[INFO] |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.10.5:compile
[INFO] +- org.apache.kafka:kafka-clients:jar:6.0.0-ccs-SNAPSHOT:compile
[INFO] |  +- com.github.luben:zstd-jni:jar:1.4.4-7:compile
[INFO] |  +- org.lz4:lz4-java:jar:1.7.1:compile
[INFO] |  +- org.xerial.snappy:snappy-java:jar:1.1.7.3:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:1.7.30:compile
[INFO] +- io.confluent:kafka-avro-serializer:jar:6.0.0-SNAPSHOT:compile
[INFO] |  +- org.apache.avro:avro:jar:1.9.2:compile
[INFO] |  |  \- org.apache.commons:commons-compress:jar:1.19:compile
[INFO] |  +- io.confluent:kafka-schema-serializer:jar:6.0.0-SNAPSHOT:compile
[INFO] |  \- io.confluent:kafka-schema-registry-client:jar:6.0.0-SNAPSHOT:compile
[INFO] |     \- javax.ws.rs:javax.ws.rs-api:jar:2.1.1:compile
[INFO] +- io.confluent:kafka-json-serializer:jar:6.0.0-SNAPSHOT:compile
[INFO] +- io.confluent:kafka-json-schema-serializer:jar:6.0.0-SNAPSHOT:compile
[INFO] |  \- io.confluent:kafka-json-schema-provider:jar:6.0.0-SNAPSHOT:compile
[INFO] |     +- com.github.everit-org.json-schema:org.everit.json.schema:jar:1.12.1:compile
[INFO] |     |  +- org.json:json:jar:20190722:compile
[INFO] |     |  +- commons-validator:commons-validator:jar:1.6:compile
[INFO] |     |  |  +- commons-digester:commons-digester:jar:1.8.1:compile
[INFO] |     |  |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO] |     |  |  \- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] |     |  +- com.damnhandy:handy-uri-templates:jar:2.1.8:compile
[INFO] |     |  \- com.google.re2j:re2j:jar:1.3:compile
[INFO] |     +- com.fasterxml.jackson.datatype:jackson-datatype-joda:jar:2.10.5:compile
[INFO] |     |  \- joda-time:joda-time:jar:2.9.9:compile
[INFO] |     +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.10.5:compile
[INFO] |     +- com.fasterxml.jackson.module:jackson-module-parameter-names:jar:2.10.5:compile
[INFO] |     +- com.kjetland:mbknor-jackson-jsonschema_2.13:jar:1.0.39:compile
[INFO] |     |  +- org.jetbrains.kotlin:kotlin-scripting-compiler-embeddable:jar:1.3.50:compile
[INFO] |     |  |  \- org.jetbrains.kotlin:kotlin-scripting-compiler-impl-embeddable:jar:1.3.50:runtime
[INFO] |     |  |     +- org.jetbrains.kotlin:kotlin-scripting-common:jar:1.3.50:runtime
[INFO] |     |  |     |  \- org.jetbrains.kotlin:kotlin-reflect:jar:1.3.50:runtime
[INFO] |     |  |     +- org.jetbrains.kotlin:kotlin-scripting-jvm:jar:1.3.50:runtime
[INFO] |     |  |     |  \- org.jetbrains.kotlin:kotlin-script-runtime:jar:1.3.50:runtime
[INFO] |     |  |     \- org.jetbrains.kotlinx:kotlinx-coroutines-core:jar:1.1.1:runtime
[INFO] |     |  +- javax.validation:validation-api:jar:2.0.1.Final:compile
[INFO] |     |  \- io.github.classgraph:classgraph:jar:4.8.21:compile
[INFO] |     \- org.jetbrains.kotlin:kotlin-stdlib:jar:1.4.0-rc:compile
[INFO] |        +- org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.4.0-rc:compile
[INFO] |        \- org.jetbrains:annotations:jar:13.0:compile
[INFO] +- io.confluent:kafka-protobuf-serializer:jar:6.0.0-SNAPSHOT:compile
[INFO] |  +- io.confluent:kafka-protobuf-provider:jar:6.0.0-SNAPSHOT:compile
[INFO] |  |  +- com.squareup.wire:wire-schema:jar:3.2.2:compile
[INFO] |  |  |  +- org.jetbrains.kotlin:kotlin-stdlib-jdk8:jar:1.3.71:compile
[INFO] |  |  |  |  \- org.jetbrains.kotlin:kotlin-stdlib-jdk7:jar:1.3.71:compile
[INFO] |  |  |  +- com.squareup.wire:wire-runtime:jar:3.2.2:compile
[INFO] |  |  |  \- com.squareup.okio:okio:jar:2.5.0:compile
[INFO] |  |  \- com.google.protobuf:protobuf-java:jar:3.11.4:compile
[INFO] |  \- com.google.protobuf:protobuf-java-util:jar:3.11.4:compile
[INFO] |     \- com.google.code.gson:gson:jar:2.8.6:compile
[INFO] +- org.slf4j:slf4j-log4j12:jar:1.7.30:compile
[INFO] +- io.confluent:confluent-log4j:jar:1.2.17-cp2:compile
[INFO] +- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.10.5:compile
[INFO] |  \- com.fasterxml.jackson.core:jackson-core:jar:2.10.5:compile
[INFO] +- com.fasterxml.jackson.datatype:jackson-datatype-guava:jar:2.10.5:compile
[INFO] +- com.google.guava:guava:jar:28.1-jre:compile
[INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  +- org.checkerframework:checker-qual:jar:2.8.1:compile
[INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.3.2:compile
[INFO] |  +- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] |  \- org.codehaus.mojo:animal-sniffer-annotations:jar:1.18:compile
[INFO] +- io.confluent:rest-utils-test:jar:6.0.0-SNAPSHOT:test
[INFO] |  +- org.glassfish.jersey.test-framework:jersey-test-framework-core:jar:2.30:test
[INFO] |  |  \- jakarta.servlet:jakarta.servlet-api:jar:4.0.3:test
[INFO] |  \- org.glassfish.jersey.test-framework.providers:jersey-test-framework-provider-jetty:jar:2.30:test
[INFO] |     \- org.glassfish.jersey.containers:jersey-container-jetty-http:jar:2.30:test
[INFO] +- junit:junit:jar:4.13:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] +- org.apache.kafka:kafka_2.13:jar:6.0.0-ccs-SNAPSHOT:test
[INFO] |  +- com.fasterxml.jackson.module:jackson-module-scala_2.13:jar:2.10.5:test
[INFO] |  |  \- com.fasterxml.jackson.module:jackson-module-paranamer:jar:2.10.5:test
[INFO] |  |     \- com.thoughtworks.paranamer:paranamer:jar:2.8:test
[INFO] |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-csv:jar:2.10.5:test
[INFO] |  +- net.sf.jopt-simple:jopt-simple:jar:5.0.4:test
[INFO] |  +- com.yammer.metrics:metrics-core:jar:2.2.0:test
[INFO] |  +- org.scala-lang.modules:scala-collection-compat_2.13:jar:2.1.6:test
[INFO] |  +- org.scala-lang.modules:scala-java8-compat_2.13:jar:0.9.1:test
[INFO] |  +- org.scala-lang:scala-library:jar:2.13.2:compile
[INFO] |  +- org.scala-lang:scala-reflect:jar:2.13.2:test
[INFO] |  +- com.typesafe.scala-logging:scala-logging_2.13:jar:3.9.2:test
[INFO] |  +- org.apache.zookeeper:zookeeper:jar:3.5.8:test
[INFO] |  |  +- org.apache.zookeeper:zookeeper-jute:jar:3.5.8:test
[INFO] |  |  +- org.apache.yetus:audience-annotations:jar:0.5.0:test
[INFO] |  |  +- io.netty:netty-handler:jar:4.1.48.Final:test
[INFO] |  |  |  +- io.netty:netty-common:jar:4.1.48.Final:test
[INFO] |  |  |  +- io.netty:netty-resolver:jar:4.1.48.Final:test
[INFO] |  |  |  +- io.netty:netty-buffer:jar:4.1.48.Final:test
[INFO] |  |  |  +- io.netty:netty-transport:jar:4.1.48.Final:test
[INFO] |  |  |  \- io.netty:netty-codec:jar:4.1.48.Final:test
[INFO] |  |  \- io.netty:netty-transport-native-epoll:jar:4.1.48.Final:test
[INFO] |  |     \- io.netty:netty-transport-native-unix-common:jar:4.1.48.Final:test
[INFO] |  \- commons-cli:commons-cli:jar:1.4:test
[INFO] +- org.apache.kafka:kafka_2.13:jar:test:6.0.0-ccs-SNAPSHOT:test
[INFO] +- org.apache.kafka:kafka-clients:jar:test:6.0.0-ccs-SNAPSHOT:test
[INFO] +- io.confluent:kafka-schema-registry:jar:6.0.0-SNAPSHOT:test
[INFO] |  +- com.101tec:zkclient:jar:0.11:test
[INFO] |  +- io.swagger:swagger-annotations:jar:1.6.2:compile
[INFO] |  \- io.swagger:swagger-core:jar:1.6.2:test
[INFO] |     +- org.apache.commons:commons-lang3:jar:3.2.1:test
[INFO] |     +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.10.5:test
[INFO] |     |  \- org.yaml:snakeyaml:jar:1.26:test
[INFO] |     \- io.swagger:swagger-models:jar:1.6.2:test
[INFO] +- org.easymock:easymock:jar:4.0.1:test
[INFO] |  \- org.objenesis:objenesis:jar:3.0.1:test
[INFO] +- io.confluent:common-utils:jar:6.0.0-SNAPSHOT:compile
[INFO] \- io.confluent:assembly-plugin-boilerplate:zip:resources:6.0.0-SNAPSHOT:provided

File: kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestApplication.java [hunk 1]
```diff
@@ -93,13 +93,11 @@ protected void setupInjectedResources(
       ProducerPool producerPool,
       KafkaConsumerManager kafkaConsumerManager
   ) {
-    if (StringUtil.isBlank(appConfig.getString(KafkaRestConfig.BOOTSTRAP_SERVERS_CONFIG))
-        && StringUtil.isBlank(appConfig.getString(KafkaRestConfig.ZOOKEEPER_CONNECT_CONFIG))) {
-      throw new RuntimeException("Atleast one of " + KafkaRestConfig.BOOTSTRAP_SERVERS_CONFIG + " "
-                                 + "or "
-                                    + KafkaRestConfig.ZOOKEEPER_CONNECT_CONFIG
-                                    + " needs to be configured");
+    /* TODO: Update EmbeddedServerTestHarness to allow required BOOTSTRAP_SERVERS_CONFIG */
```
Comment (ijuma) [inline kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestApplication.java:96]: Let's not have TODOs in the code please. If this needs to be done as a follow-up, we should file a JIRA and track it that way.

File: kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java [hunk 2]
```diff
@@ -202,18 +183,12 @@
...
-   */
-  public static final String KAFKACLIENT_ZK_SESSION_TIMEOUT_MS_CONFIG
-      = "client.zk.session.timeout.ms";
   public static final String KAFKACLIENT_TIMEOUT_CONFIG = "client.timeout.ms";
   /**
    * <code>client.init.timeout.ms</code>
    */
   public static final String KAFKACLIENT_INIT_TIMEOUT_CONFIG = "client.init.timeout.ms";
 
-  public static final String ZOOKEEPER_SET_ACL_CONFIG = "zookeeper.set.acl";
```
Comment (ijuma) [inline kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java:216]: This was unused, right?

Comment (rnpridgeon) [reply kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java:216]: It was, no need to keep it around though right?

File: kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestApplication.java (hunk 1 above)
Comment (rnpridgeon) [reply kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestApplication.java:96]: Ack, will remove.

File: kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java (hunk 2 above)
Comment (ijuma) [reply kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java:216]: Yes, we can remove.

**Comment by ijuma** [comment]
Let's update the PR description and commit text to include the additional changes in the PR.

**Pull request merged**